packages/contracts/contracts/CdpManager.sol

```diff
@@ -407,6 +403,11 @@ contract CdpManager is CdpManagerStorage, ICdpManager, Proxy {
                 // therefore we could not redeem from the last Cdp
                 if (singleRedemption.cancelledPartial) break;
 
+                // prepare for reinsertion if there is partial redemption
+                if (singleRedemption.newPartialNICR > 0) {
```

Comment:
If the partial has happened this will be non-zero, it will use that as flag that the partial requires re-insertion

NOTE: The partial is still cancelled if the NICR hint is wrong, so this avoids massive gas expenditures